src/Microsoft.DotNet.Build.Tasks/PackageFiles/CodeCoverage.targets

```diff
@@ -131,6 +91,38 @@
   </Target>
   <!-- *********************************************************************************************** -->
 
+  <Target Name="CreateCoverageDedicatedRuntime"
+    BeforeTargets="GenerateTestExecutionScripts"
+    DependsOnTargets="GenerateWindowsPdbsForAssemblyBeingTested"
+    Condition="'$(CodeCoverageEnabled)'=='true' and '$(UseCoverageDedicatedRuntime)'=='true'">
+    <PropertyGroup>
+      <CoverageDedicatedRuntimeDir>$(TestHostRootPath)shared/Microsoft.NETCore.App/10.10.10</CoverageDedicatedRuntimeDir>
```

Comment:
>10.10.10 [](start = 83, length = 8)

do we'll need to update this version every while?

Reply:
how this will work with the private runs? I mean when someone uses /p:CoreClrOverridePath?

---
In reply to: [197517955](https://github.com/dotnet/buildtools/pull/2077#discussion_r197517955) [](ancestors = 197517955)

Reply:
Is there anyway we can detect the version of the used corelib automatically? instead of hardcoding version values? I am not expert here but I can imagine we can get what we need from the runtime folder.

Reply:
This is the behavior of runtimeconfig.json when we use dotnet.exe, AFAIK we have to specify it in the file and then it will look in the respective folder.